server/src/main/java/io/deephaven/server/plugin/python/PyObjectRefCountedNode.java

```diff
@@ -0,0 +1,40 @@
+package io.deephaven.server.plugin.python;
+
+import io.deephaven.engine.liveness.LivenessArtifact;
+import io.deephaven.engine.liveness.ReferenceCountedLivenessNode;
+import io.deephaven.util.annotations.ScriptApi;
+import org.jpy.PyObject;
+
+/**
+ * Provides a mapping between Python refcount and Deephaven's liveness mechanism, allowing liveness scopes to manage the
+ * single <a href='https://docs.python.org/3/glossary.html#term-strong-reference'>strong reference</a> that the PyObject
+ * instance represents. This way, as long as PyObjectRefCountedNode instances are managed correctly by their parent
+ * scope, the PyObject strong reference will be correctly released when the object is no longer used.
+ */
+public class PyObjectRefCountedNode extends LivenessArtifact {
+    private final PyObject pythonObject;
+
+    /**
+     * Constructs a PyObjectRefCountedNode instance by wrapping a PyObject. The caller should ensure that an appropriate
+     * liveness scope is open to retain this instance.
+     */
+    @ScriptApi // Called by internal Python code
+    public PyObjectRefCountedNode(PyObject pythonObject) {
+        super(true);
+        this.pythonObject = pythonObject;
+    }
+
+    @Override
+    protected void destroy() {
+        super.destroy();
+        pythonObject.close();
+    }
```

Comment:
`PyObject#close` is an explicit way for us to say "we aren't going to use this PyObject instance from java anymore" (ie, we don't plan on keeping any hard references to it); there is definitely reason for it to exist IMO. User / implementation code should proactively `PyObject#close` when appropriate (and there are definitely cases where we could be better about calling close), even though yes, it may block on the GIL... we should never do this in destroy() though...

Reply:
Does `destroy()` not have the same same semantics, the object will go out of scope, we don't expect any remaining hard references? This isn't related to `finalize()`, and should happen well before finalize goes off (or not at all, I don't see any Cleaner/PhantomReference/finalize implementations that call `destroy()`). 

The alternative is a patch that "fixes" 1775 by wrapping PyObject in some other generic Object and just leaking the references, and hoping that making Python GC dependent on Java GC won't ever go wrong.

Reply:
Ohhhhhh I had a brain fart and was thinking this was `finalize()` the whole time.

Reply:
Haha yeah I'd be pretty unhappy with this patch then too, 100%. Lmk if you'd want a call or some time to debug to make sure about this.